src/index.js

```diff
@@ -9,13 +9,21 @@ export default function factory({
   onError = onerror,
   onNoMatch = onerror.bind(null, { status: 404, message: "not found" }),
   attachParams = false,
+  disableResponseWait = false,
 } = {}) {
-  function nc(req, res) {
+  async function nc(req, res) {
+    let closeP;
+    if (!disableResponseWait && "once" in res)
+      closeP = new Promise((resolve) => {
+        res.once("close", resolve);
+        if (isResSent(res)) resolve();
```

Comment:
I'm seeing this setup happening before my route code executes, and at that point the response hasn't been sent. I'm not sure this line does anything.

Reply:
There might be a chance that the user finishes the response before actually using our handler. It's definitely rare, but I just add the check for completion.

```js
const handler = nc();

export default function (req, res) {
   res.end();
   await handler(req,res);
}

```